bigtable-hbase-parent/bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableConnection.java

```diff
@@ -73,6 +74,11 @@ public void run() {
       }
     };
     Runtime.getRuntime().addShutdownHook(new Thread(shutDownRunnable));
+
+    // Force the loading of HConstants.class, which shares a bi-directional reference with KeyValue.
+    // This bi-drectional relationship causes problems when KeyVale and HConstants are class loaded
+    // in different threads.  This forces a clean class loading of both classes.
+    HConstants.class.getName();
```

Comment:
I can't think of another bidirectional dependency cycle. But I think that the static initializer in Adapters has a fairly large graph of dependencies that we will have to be loaded at some point. By eagerly loading Adapters in the user's main thread we are drastically lowering our chances of another deadlock due to multiple threads contending for classloading locks.